vm/vm.go

```diff
@@ -50,8 +55,11 @@ func (m *vm) Invoke(name string, arguments ...interpreter.Value) (interpreter.Va
 	}
 
 	res, err := f.Call(m.store, rawArguments...)
-	if err != nil {
-		return nil, err
+
+	var trap *wasmtime.Trap
+
+	if errors.As(err, trap) {
+		return nil, trap
```

Comment:
Why are only traps specifically detected and rejected? What about other kinds of errors?